src/models/responses.py

```diff
@@ -66,13 +88,48 @@ class QueryResponse(BaseModel):
         ],
     )
 
+    rag_chunks: Optional[List[RAGChunk]] = Field(
```

Comment:
use native type hint for list. same for any other places..
```suggestion
    rag_chunks: Optional[list[RAGChunk]] = Field(
```

Reply:
IMHO these lists are not optional - these are filled in by previous code from this PR. Probably make them non-optional with default value set to `[]`? WDYT?

Reply:
+1